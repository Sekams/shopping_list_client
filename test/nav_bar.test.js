import React from 'react';
import NavBar from '../src/components/nav_bar'

test("renders the nav_bar component", () => {
    const wrapper = shallow(
        <NavBar />
    );
    expect(wrapper).toMatchSnapshot();
});

test("calls the onSearchTermChanged", () => {
    const props = {
        onSearchTermChange: jest.fn(),
    };
    const wrapper = shallow(
        <NavBar {...props} />
    );
    wrapper.props().children.props.children[1].props.children.props.onSearchTermChange()
    expect(wrapper).toMatchSnapshot();
});