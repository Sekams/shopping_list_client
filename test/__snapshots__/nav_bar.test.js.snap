// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls the onSearchTermChanged 1`] = `
<nav
  className="navbar navbar-fixed-top drop-shadow"
  id="home-navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <Link
        className="navbar-brand"
        replace={false}
        to="/"
      >
        Shopping List
      </Link>
    </div>
    <div
      className="collapse navbar-collapse"
    >
      <NavBarOptions
        onSearchTermChange={[Function]}
      />
    </div>
  </div>
</nav>
`;

exports[`renders the nav_bar component 1`] = `
<nav
  className="navbar navbar-fixed-top drop-shadow"
  id="home-navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-header"
    >
      <Link
        className="navbar-brand"
        replace={false}
        to="/"
      >
        Shopping List
      </Link>
    </div>
    <div
      className="collapse navbar-collapse"
    >
      <NavBarOptions
        onSearchTermChange={[Function]}
      />
    </div>
  </div>
</nav>
`;
